src/assembly/types.ts

```diff
@@ -6,7 +6,16 @@ export type StackAsset = FileManifestEntry | DockerImageManifestEntry;
 /**
  * Map of CDK construct path to logicalId
  */
-export type StackMetadata = { [path: string]: string };
+export type StackMetadata = { [path: string]: StackAddress };
+
+/**
+ * StackAddress uniquely identifies a resource in a cloud assembly
+ * across several (nested) stacks.
+ */
+export type StackAddress = {
```

Comment:
Perhaps ResourceAddress? StackAddress makes it sound like it identifies a Stack. Doc comments on props are also nice, with examples.